rust/src/channel/mod.rs

```diff
@@ -11,42 +11,42 @@
 //! This module provides a way to define an OSDP channel and export it to
 //! LibOSDP.
 
+#[cfg(feature = "std")]
 pub mod unix_channel;
+#[cfg(feature = "std")]
 pub use unix_channel::UnixChannel;
 
-use crate::osdp_sys;
+use crate::{osdp_sys, Mutex};
```

Comment:
Shouldn't we have an explicit

```
#[cfg(feature = "std")]
use parking_lot::Mutex;
#[cfg(not(feature = "std"))]
use spin::Mutex;
```
insread of importing from crate::? It's confusing as it appears as though we are providing a mutex type.

Reply:
You are right. This feels wrong. I wanted to have the `Mutex `selection only in one place, but I agree this was not the best solution.

I solved this now, by using `Arc` instead of `Mutex` in `lib.rs`.